amundsen_application/static/js/ducks/search/reducer.ts

```diff
@@ -44,10 +58,26 @@ const initialState: SearchReducerState = {
 };
 
 export default function reducer(state: SearchReducerState = initialState, action: SearchReducerAction): SearchReducerState {
+  let newState = action.payload;
   switch (action.type) {
-    case ExecuteSearch.SUCCESS:
-      return action.payload;
-    case ExecuteSearch.FAILURE:
+    // SearchAll will reset all resources with search results or the initial state
+    case SearchAll.SUCCESS:
+      return {
+        searchTerm: newState.searchTerm,
+        dashboards: newState.dashboards || initialState.dashboards,
+        users: newState.users || initialState.users,
+        tables: newState.tables || initialState.tables,
+      };
+    // SearchResource will set only a single resource and preserves search state for other resources
```

Comment:
Why not have the payload of `SearchResource.SUCCESS` reflect that by returning the resource name and results, and then just updating that one piece of state?

Reply:
I see. I was trying to convert `search_term` to `searchTerm` while passing the rest of the fields along. If I leave `search_term` as is and pass along the response object I can just spread the new object and overwrite just the things that are included.